promoted_python_delivery_client/client/delivery_request_state.py

```diff
@@ -19,3 +19,9 @@ def get_request_to_send(self, max_request_insertions: int) -> Request:
         if req.insertion_matrix and len(req.insertion_matrix) > max_request_insertions:
             req.insertion_matrix = req.insertion_matrix[0:max_request_insertions]
         return req
+
+
+def _validate_response(response: Response) -> Response:
+    if not response.request_id:
+        raise ValueError(f"Delivery Response must have request_id; response={response}")
```

Comment:
Considering we're supposed to populate this in delivery, does this make sense to raise to the user? Or is the user still expected to do an initial population in the case that the SDK does the responding for us?

Reply:
If they get this error, it probably means that they are hitting a bad URL or we have an unsupported Delivery API response.  If you want, I could change the error message to indicate that.  LMK.

The SDK fallback sets Response.request_id.  This code path isn't executed though.